chain/src/main/scala/org/bitcoins/chain/blockchain/Blockchain.scala

```diff
@@ -53,28 +72,41 @@ object Blockchain extends BitcoinSLogger {
       blockchains =>
         val nested: Vector[Future[BlockchainUpdate]] = blockchains.map {
           blockchain =>
-            val tip = blockchain.tip
-            logger.debug(
-              s"Attempting to add new tip=${header.hashBE.hex} with prevhash=${header.previousBlockHashBE.hex} to chain with current tips=${tip.hashBE.hex}")
-            val tipResultF = TipValidation.checkNewTip(newPotentialTip = header,
-                                                       currentTip = tip,
-                                                       blockHeaderDAO =
-                                                         blockHeaderDAO)
-
-            tipResultF.map { tipResult =>
-              tipResult match {
-                case TipUpdateResult.Success(headerDb) =>
-                  val newChain =
-                    Blockchain.fromHeaders(headerDb +: blockchain.headers)
-                  BlockchainUpdate.Successful(newChain, headerDb)
-                case fail: TipUpdateResult.Failure =>
-                  BlockchainUpdate.Failed(blockchain, header, fail)
-              }
+            val prevBlockHeaderOpt =
+              blockchain.find(_.hashBE == header.previousBlockHashBE)
```

Comment:
When we're looking for block headers to connect to we're more likely to find a matching tip towards the end than the start. Would it make sense to do `blockchain.reverseIterator.find(...)`?

Reply:
Actually headers are prepended here: https://github.com/bitcoin-s/bitcoin-s/blob/master/chain/src/main/scala/org/bitcoins/chain/blockchain/Blockchain.scala#L68